packages/@yoda/bluetooth/hfp.js

```diff
@@ -0,0 +1,275 @@
+ *   console.log(`bluetooth state: ${state}`)
+ * })
+ * hfp.open()
+ */
+BluetoothHfp.prototype.open = function () {
+  logger.debug(`open()`)
+  var msg = {
+    name: this.localName,
+    unique: false
+  }
+  this._send('ON', msg)
+}
+
+/**
+ * Turn off the bluetooth.
+ *
+ * You can listen following changed state events:
+ * - `protocol.RADIO_STATE.OFF` when bluetooth is closed.
+ *
+ * @returns {null}
+ * @fires module:@yoda/bluetooth/BluetoothHfp#radio_state_changed
+ */
+BluetoothHfp.prototype.close = function () {
+  logger.debug(`close(${this.lastMsg.hfpstate})`)
+  if (this.lastMsg.hfpstate === 'closed') {
+    logger.warn('close() while last state is already closed.')
+  }
+  this._send('OFF')
+}
+
+/**
+ * Answer incoming call.
+ * @return {null}
+ * @fires module:@yoda/bluetooth/BluetoothHfp#call_state_changed
+ */
+BluetoothHfp.prototype.answer = function () {
+  logger.debug(`answer()`)
+  this._send('ANSWERCALL')
+}
+
+/**
+ * Hang up call.
+ * @return {null}
+ * @fires module:@yoda/bluetooth/BluetoothHfp#call_state_changed
+ */
+BluetoothHfp.prototype.hangup = function () {
+  logger.debug(`hangup()`)
+  this._send('HANGUP')
+}
+
+/**
+ * Outgoing a call.
+ * @param {string} number - Specify the destination phone number.
+ * @return {null}
+ * @fires module:@yoda/bluetooth/BluetoothHfp#call_state_changed
+ */
+BluetoothHfp.prototype.dial = function (number) {
+  logger.debug(`dial(${number})`)
+  this._send('DIALING', {'NUMBER': number})
+}
+
+/**
+ * Get if bluetooth is opened.
+ * @returns {boolean} - `true` if bluetooth is opened else `false`.
+ */
+BluetoothHfp.prototype.isOpened = function () {
+  return this.lastMsg.hfpstate === 'opened'
+}
+
+/**
+ * Get if this device is connected with remote device.
+ * @returns {boolean} - `true` if blueooth is connected with remote device else `false`.
+ */
+BluetoothHfp.prototype.isConnected = function () {
+  return this.lastMsg.connect_state === 'connected'
+}
+
+/**
+ * @typedef {object} BluetoothDevice
+ * @property {string} name - The device's name.
+ * @property {string} address - The device's MAC address.
+ */
+
+/**
+ * Get connected bluetooth device.
+ * @returns {BluetoothDevice|null} - Current connected bluetooth device object or `null` if no connected device.
+ */
+BluetoothHfp.prototype.getConnectedDevice = function () {
+  if (!this.isConnected()) {
+    return null
+  } else {
+    return {
+      address: this.lastMsg.connect_address,
+      name: this.lastMsg.connect_name
+    }
+  }
+}
+
+/**
+ * Get if is calling.
+ * @returns {boolean} - `true`if is calling else `false`.
+ */
+BluetoothHfp.prototype.isCalling = function () {
+  return this.lastMsg.call === 'active'
+}
+
+/**
+ * Destroy bluetooth profile adapter, thus means bluetooth will always be turned `OFF` automatically.
+ */
+BluetoothHfp.prototype.destroy = function () {
+  var destroyFunc = function () {
+    this.removeAllListeners()
+    this._flora.destruct()
+    this._end = true
+  }.bind(this)
```

Comment:
Use arrow function to instead the `bind(this)`.

Reply:
Changed to arrow function.